src/can.c

```diff
@@ -109,8 +178,41 @@ static void thingset_can_report_rx_cb(const struct device *dev, struct can_frame
     const struct thingset_can *ts_can = user_data;
     uint16_t data_id = THINGSET_CAN_DATA_ID_GET(frame->id);
     uint8_t source_addr = THINGSET_CAN_SOURCE_GET(frame->id);
-
-    ts_can->report_rx_cb(data_id, frame->data, can_dlc_to_bytes(frame->dlc), source_addr);
+#ifdef CONFIG_THINGSET_CAN_PACKETIZED_REPORTS_RX
+    if (THINGSET_CAN_PACKETIZED_REPORT(frame->id)) {
+        struct net_buf *buffer = NULL;
+        if ((buffer = thingset_can_get_rx_buf(source_addr)) != NULL) {
```

Comment:
Why not like this? IMO more clear and eliminates the redundant assignment.

```suggestion
        struct net_buf *buffer = thingset_can_get_rx_buf(source_addr);
        if (buffer != NULL) {
```

Reply:
Yes, fair. I think this is a legacy of another approach, because it doesn't really make sense now you mention it.